lightning-invoice/src/utils.rs

```diff
@@ -489,4 +697,115 @@ mod test {
 			_ => panic!("Unexpected event")
 		}
 	}
+
+    #[test]
+    fn test_create_invoice_with_description_hash() {
+        let chanmon_cfgs = create_chanmon_cfgs(2);
+        let node_cfgs = create_node_cfgs(2, &chanmon_cfgs);
+        let node_chanmgrs = create_node_chanmgrs(2, &node_cfgs, &[None, None]);
+        let nodes = create_network(2, &node_cfgs, &node_chanmgrs);
+
+        let invoice = ::utils::create_invoice_from_channelmanager_with_description_hash_and_duration_since_epoch(
+            &nodes[1].node,
+            nodes[1].keys_manager,
+            Currency::BitcoinTestnet,
+            Some(10_000),
+            "Testing description_hash".to_string(),
+            Duration::from_secs(1234567),
+        )
+        .unwrap();
+        assert_eq!(invoice.amount_pico_btc(), Some(100_000));
+        assert_eq!(
+            invoice.min_final_cltv_expiry(),
+            MIN_FINAL_CLTV_EXPIRY as u64
+        );
+        assert_eq!(
+            invoice.description(),
+            InvoiceDescription::Hash(&crate::Sha256(Sha256::hash(
+                "Testing description_hash".as_bytes()
+            )))
+        );
+	}
+
+
+	#[test]
+    #[cfg(feature = "std")]
+    fn create_phantom_invoice_with_description_hash() {
+        let mut chanmon_cfgs = create_chanmon_cfgs(3);
+        let seed_1 = [42 as u8; 32];
+        let seed_2 = [43 as u8; 32];
+        let cross_node_seed = [44 as u8; 32];
+        chanmon_cfgs[1].keys_manager.backing =
+            PhantomKeysManager::new(&seed_1, 43, 44, &cross_node_seed);
+        chanmon_cfgs[2].keys_manager.backing =
+            PhantomKeysManager::new(&seed_2, 43, 44, &cross_node_seed);
+        let node_cfgs = create_node_cfgs(3, &chanmon_cfgs);
+        let node_chanmgrs = create_node_chanmgrs(3, &node_cfgs, &[None, None, None]);
+        let nodes = create_network(3, &node_cfgs, &node_chanmgrs);
+        let chan_0_1 = create_announced_chan_between_nodes_with_value(
+            &nodes,
+            0,
+            1,
+            100000,
+            10001,
+            InitFeatures::known(),
+            InitFeatures::known(),
+        );
+        nodes[0]
+            .node
+            .handle_channel_update(&nodes[1].node.get_our_node_id(), &chan_0_1.1);
+        nodes[1]
+            .node
+            .handle_channel_update(&nodes[0].node.get_our_node_id(), &chan_0_1.0);
+        let chan_0_2 = create_announced_chan_between_nodes_with_value(
+            &nodes,
+            0,
+            2,
+            100000,
+            10001,
+            InitFeatures::known(),
+            InitFeatures::known(),
+        );
+        nodes[0]
+            .node
```

Comment:
I know this is the rustfmt way and many rust projects use this kind of super-vertical layout, but I find it completely unreadable (and its the primary reason we don't use rustfmt here and I believe the primary reason rust-bitcoin doesn't use it either). If possible, could you condense lines a bit so that its easier for the eye to scan between statements? Only break lines if the line is reaching 100 chars in length, roughly.